### PR TITLE
misc:update packaging tool,add pre-check scripts/ add patch function …

### PR DESCRIPTION
**Pull request by zhanqigithub**
…/optimize

Tracked-On: #5022

Signed-off-by: zhanqi <sherry.qi.zhang@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: misc/packaging/acrn-hypervisor.preinst [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+#* Copyright (c) 2020 Intel Corporation All rights reserved.
```
Comment (dbkinder) [inline misc/packaging/acrn-hypervisor.preinst:2]: delete "all rights reserved"
add ``SPDX-License-Identifier: BSD-3-Clause``

File: misc/packaging/acrn-hypervisor.postinst [hunk 2]
```diff
@@ -1,68 +1,107 @@
 #!/bin/bash
 #* Copyright (c) 2020 Intel Corporation All rights reserved.
```
Comment (dbkinder) [inline misc/packaging/acrn-hypervisor.postinst:2]: delete "all rights reserved"
add ``SPDX-License-Identifier: BSD-3-Clause``

File: misc/packaging/release.json [hunk 3]
```diff
@@ -1,68 +1,64 @@
...
+		},
+		"release":"0"
+	},
+	"acrn_patch":
+	{
+		"patch_need":"false",
+		"patch_dir":"acrn_additional_patch",
+		"patch_list":
+		{
+			"0001-xxxxxxxx.patch": "true",
```
Comment (dbkinder) [inline misc/packaging/release.json:40]: I guess these are placeholders for patches that don't currently exist?

File: misc/packaging/acrn-hypervisor.postinst [hunk 4]
```diff
@@ -1,68 +1,107 @@
...
 # postinst script for acrn-hypervisor
+# please NOTE ！！！  senario_info/board_info changed by python scripts, so do not add content there!!!
```
[the remark this answers is not in the document]
Comment (zhanqigithub) [reply misc/packaging/acrn-hypervisor.postinst:4]: changed

File: misc/packaging/acrn-hypervisor.postinst [hunk 5]
```diff
@@ -1,68 +1,107 @@
...
+    let s_num=i+1
+    echo  $s_num ${scenario_info[i]};
+
+}
+
+
+while :
+do
+    read num
+    if [ $num = 0 ] || [ $num -gt ${scenario_num} ]; then
```
[the remark this answers is not in the document]
Comment (zhanqigithub) [reply misc/packaging/acrn-hypervisor.postinst:31]: next version will add strong error handling state

File: misc/packaging/acrn-hypervisor.postinst [hunk 6]
```diff
@@ -1,68 +1,107 @@
...
-if [[ $num3 -eq 1 ]]
-then
-        echo "disk type is nvme"
-        type="nvme"
-elif [[ $num3 -eq 2 ]]
-then
-        echo "disk type is sda"
-        type="sda"
-fi
+disk_info=(nvme sda)
```
[the remark this answers is not in the document]
Comment (zhanqigithub) [reply misc/packaging/acrn-hypervisor.postinst:79]: disk info this version is not auto generate, so separate from  scenario, next version we may add board  info handling

File: misc/packaging/acrn-hypervisor.preinst (hunk 1 above)
Comment (zhanqigithub) [reply misc/packaging/acrn-hypervisor.preinst:2]: changed

File: misc/packaging/acrn-hypervisor.postinst (hunk 2 above)
Comment (zhanqigithub) [reply misc/packaging/acrn-hypervisor.postinst:2]: changed

File: misc/packaging/release.json (hunk 3 above)
Comment (zhanqigithub) [reply misc/packaging/release.json:40]: yes, patch should provide by user

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**